x/concentrated-liquidity/swapstrategy/swap_strategy_test.go

```diff
@@ -271,3 +272,90 @@ func (suite *StrategyTestSuite) TestGetPriceLimit() {
 		})
 	}
 }
+
+// validates that the correct sqrt price limits are returned
+// as dictated by the GetSqrtPriceLimit spec.
+// Tests that the correct sqrt function is used depending
+// on the value of the priceLimit input. Below min, the big dec sqrt is used.
+// Above min, the dec sqrt is used.
+func (suite *StrategyTestSuite) TestGetSqrtPriceLimit() {
+	var (
+		oneULPUnderThreshold = types.MinSpotPriceBigDec.Sub(oneULPBigDec)
+		atThreshold          = types.MinSpotPriceBigDec
+		oneULPAboveThreshold = types.MinSpotPriceBigDec.Add(oneULPBigDec)
+	)
```

Comment:
Note to capture an offline discussion we had for posterity: this threshold does not break monotonicity because it is at a value that is equivalent in regular `Dec` and `BigDec` domains.

Specifically, calling the conversion of the threshold spot price from `BigDec` to `Dec` is a misnomer because nothing is actually being "truncated" (which would break monotonicity) – the padded 18 zeroes are simply being trimmed

Reply:
It might be worthwhile to capture this in comments somewhere if it isn't already (sorry if it is and I missed it)

Reply:
This is unrelated to testing the monotonicity of tick-to-sqrt price conversions that can be found in the rest of the test suite. It simply tests the execution flow of `GetSqrtPriceLimit`.

Added clarifying comment: [e46ef8a](https://github.com/osmosis-labs/osmosis/pull/6368/commits/e46ef8ab6ad259453aacf3c5f95d4d09153b894f)